File: DSA/Java/tests/dsa/lib/algo/sort/BubbleSortTest.java
package dsa.lib.algo.sort;

import org.junit.Test;

import static org.junit.Assert.*;

public class BubbleSortTest {
    @Test
    public void oneNum() {
        int[] ary = {1};
        BubbleSort.sort(ary);
        assertArrayEquals(new int[]{1}, ary);
    }

    @Test
    public void case1() {
        int[] ary = {4, 5, 6, 3, 2, 1};
        BubbleSort.sort(ary);
        assertArrayEquals(new int[]{1, 2, 3, 4, 5, 6}, ary);
    }
}

File: DSA/Cpp/test/dsa/lib/algo/sort/bubble_sort_test.cpp
#include <dsa/lib/algo/sort/bubble_sort.hpp>
#include "sort_test.h"
namespace dsa::lib::algo::sort {
namespace {

class BubbleSortTest : public SortTest {};
TEST_F(BubbleSortTest, oneNum) {
    int ary[] = {1};
    int expected[] = {1};
    BubbleSort::sort(ary, arylen(ary));
    EXPECT_ARRAY_EQ(expected, arylen(expected), ary, arylen(ary));
}
TEST_F(BubbleSortTest, case1) {
    int ary[] = {4, 5, 6, 3, 2, 1};
    int expected[] = {1, 2, 3, 4, 5, 6};
    BubbleSort::sort(ary, arylen(ary));
    EXPECT_ARRAY_EQ(expected, arylen(expected), ary, arylen(ary));
}

}
};